python/ray/llm/_internal/serve/deployments/llm/vllm/kv_transfer_backends/base.py

```diff
@@ -35,6 +35,38 @@ def _get_unique_suffix(self, len: int = 6) -> str:
         """
         return "".join(random.choices(string.ascii_letters + string.digits, k=len))
 
+    def _compute_port_offset(self) -> int:
+        """Compute a deterministic port offset for this replica/process.
+
+        Priority:
+        1) data_parallel_rank if present (set by DPServer).
+        2) Stable hash of Serve replica tag (avoids cross-replica collisions when TP/PP only).
+
+        Returns:
+            A small non-negative integer to add to a base port.
+        """
+        # Prefer explicit DP rank when available
+        dp_rank = self.llm_config.engine_kwargs.get("data_parallel_rank")
+        if isinstance(dp_rank, int) and dp_rank >= 0:
+            return dp_rank
+
+        # Fall back to a stable hash of the Serve replica tag if available
+        try:
+            # Import locally to avoid import-time side effects
+            from ray import serve  # type: ignore
+
+            rc = serve.get_replica_context()
+            if rc and getattr(rc, "replica_tag", None):
+                import zlib
```

Comment:
is this an extra dependency? We should import these things at top so that they raise errors on ci/cd tests.